doc/planning.md

```diff
@@ -0,0 +1,113 @@
+# gRPC 5.0 planning
+
+| Legend |                          |
+|--------|--------------------------|
+| 🔴      | Stretch goal |
```

Comment:
nit: are these symbols some standard you're using for ASP.NET core planning? They seem a bit unnecessary to me, just `[stretch]` prefix would probably suffice. Leaving up to you.